api/v1/routes/categories.py

```diff
@@ -0,0 +1,23 @@
+from fastapi import APIRouter, Depends, Query
+from sqlalchemy.orm import Session
+from typing import Optional
+from ..schemas.category import CategoryList
+from ..services import category as category_service
+from ...db.database import get_db
+from ..services.cache import cache
+
+# from ..auth.jwt import get_current_user    (Add your jwt auth to get current logged in user)
+
+router = APIRouter()
+
+@router.get("/categories", response_model=CategoryList)
+@cache
+async def read_categories(
```

Comment:
You don't need to create a router for this, just use the product router you created